src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesSignedCommits.java

```diff
@@ -0,0 +1,209 @@
+
+    return this;
+  }
+
+  /**
+   * It gathers the list of commits within one year from the current date. Then it calculates
+   * density ratio. The density ratio is the percentage calculation of number of verified signed
+   * commits against the total number of commits in the year. If the density ratio in % crosses a
+   * certain percentage threshold ({@link #SIGNED_COMMIT_PERCENTAGE_THRESHOLD}), then it says that
+   * the project uses verified signed commits. Otherwise, it says that the project does not use
+   * verified signed commits.
+   * 
+   * @return {@link BooleanValue} value of the feature {@link #USES_VERIFIED_SIGNED_COMMITS}.
+   */
+  private Value<Boolean> usesSignedCommits() {
+    double signedCounter = 0;
+    double counter = 0;
+    Date date = new Date(System.currentTimeMillis() - DELTA);
+    String commitDate = DATE_FORMAT.format(date);
+    int pageNumber = FIRST_PAGE;
+
+    // TODO: The data gathering process here from GitHub API, will have to be merged with other
+    // related data providers. More information can be found in
+    // https://github.com/SAP/fosstars-rating-core/issues/69
+    try {
+      while (hasNextPage) {
+
+        JsonNode commitNodeList = commitInfo(commitDate, pageNumber);
+        if (commitNodeList == null || commitNodeList.size() == 0) {
+          break;
+        }
+        pageNumber++;
+
+        for (JsonNode commitNode : commitNodeList) {
+          counter++;
+          JsonNode verifiedNode = commitNode.findPath("verified");
+          if (verifiedNode != null && verifiedNode.booleanValue()) {
+            signedCounter++;
+          }
+        }
+      }
+    } catch (IOException e) {
+      System.out.printf("[x] Couldn't fetch data from api.github.com!%n");
+      System.out.printf("[x]     %s%n", e);
+      return UnknownValue.of(USES_VERIFIED_SIGNED_COMMITS);
+    }
+
+    if (counter > 0 && signedCounter > 0
+        && (signedCounter * 100 / counter) >= SIGNED_COMMIT_PERCENTAGE_THRESHOLD) {
+      return new BooleanValue(USES_VERIFIED_SIGNED_COMMITS, true);
+    }
+    return new BooleanValue(USES_VERIFIED_SIGNED_COMMITS, false);
+  }
+
+  /**
+   * Does a REST API call to public URL <a link="https://api.github.com">GitHub API</a> to list all
+   * the commits after a specific date.
+   * 
+   * @param date Only commits after this date will be returned. 
+   *             This is a timestamp in ISO 8601, format: YYYY-MM-DDTHH:MM:SSZ
+   * @param page Index of the page
+   * @return The {@link JsonNode} object
+   * @throws IOException #{@link HttpClient} may throw an exception during REST call.
+   * @see: GitHub Issue: <a link="https://github.com/github-api/github-api/issues/737">Get commit or
+   *       tag signature verified flag</a> and the associated
+   *       <a link= "https://github.com/github-api/github-api/pull/738">Pull Request</a>.
+   */
+  private JsonNode commitInfo(String date, int page) throws IOException {
```

Comment:
Would you mind returning `httpResponse` from this method? That would allow dropping the `hasNextPage` field (which should be at most a local variable).